keps/sig-node/2400-node-swap/README.md

```diff
@@ -441,6 +468,25 @@ Additionally, end user may decide to disable swap completely for a Pod or a cont
 In the early release of this feature, there was a goal to support cgroup v1. As the feature progressed, sig-node realized that supporting swap with cgroup v1 would be very difficult.
 Therefore, this feature is limited to cgroupv2 only. The main goal is to deprecate cgroupv1 eventually so this should not be a major inconvience.
 
+#### Memory-backed volumes
+
+Kubernetes guarantees that some volumes' memory would never reside on disk, e.g. Secrets, memory-backed emptyDirs, etc.
+Behind the scenes, Kubelet mounts such volumes as tmpfs volumes on the host.
+
+To address this risk, if `--fail-swap-on=false`, the [tmpfs noswap option](https://www.kernel.org/doc/html/latest/filesystems/tmpfs.html)
+will be used in order to prevent the volumes' pages from swapping to disk.
+
+Bear in mind that the tmpfs noswap option is fairly new and is supported in kernel versions >= 6.4. However, different
+Linux distributions can decide to backport this options to older versions of the kernel. Therefore, when
+`--fail-swap-on=false` is being provided on a node:
+* If the kernel version equals or is above 6.4, the tmpfs noswap option is being used when necessary.
+* Else, kubelet would try to mount a dummy volume with the tmpfs noswap option to understand whether the option is
+  backported. If the mount succeeds, the tmpfs noswap option is being used when necessary.
+* Else, kubelet would raise a warning about the option not being supported and the possible risk.
+
```

Comment:
- Will we document and announce the change from “Kubernetes guarantees that some volumes' memory would never reside on disk, e.g. Secrets, memory-backed emptyDirs, etc” to “Kubernetes makes a best effort to ensure”

We have the option of failing the Pod once the risk of paging out confidential data is detected. I think that should be the default with a kubelet configuration to emit an Event rather than fail the Pod. Failing the Pod by default maintains that guarantee. The kubelet could even specially taint nodes that don't provide the usual guarantee, requiring Pods to tolerate it.

Reply:
This issue had been brought up in sig-node meetings and was discussed multiple times in the past. For more information I recommend looking at https://github.com/kubernetes/kubernetes/issues/105978, and especially [this comment](https://github.com/kubernetes/kubernetes/issues/105978#issuecomment-2078246497).

Long story short, I (and many others) think that documenting this small and temporary limitation around swap is enough. In a short while the tmpfs noswap option would be very widely supported in the vast majority of the environments. So I don't think that kubernetes is truly changing its guarantees, but rather enabling swap brings a security limitation that should indeed be properly documented.